locksmithctl/daemon.go

```diff
@@ -309,8 +313,15 @@ func runDaemon() int {
 		strategy = StrategyBestEffort
 	}
 
+	// XXX: complain loudly if besteffort is used
+	if strategy == StrategyBestEffort {
+		dlog.Errorf("Reboot strategy %q is deprecated and will be removed in the future.", strategy)
+		dlog.Errorf("Please explicitly set the reboot strategy to one of %v", []string{StrategyOff, StrategyReboot, StrategyEtcdLock})
+		dlog.Error("See https://coreos.com/os/docs/latest/update-strategies.html for details on configuring reboot strategies.")
+	}
+
 	if strategy == StrategyOff {
-		fmt.Fprintf(os.Stderr, "Reboot strategy is %q - shutting down.\n", strategy)
+		dlog.Warningf("Reboot strategy is %q - shutting down.", strategy)
```

Comment:
Locksmith actually starts before coreos-cloudinit since it isnt ordered
after any config target. :(
On Feb 10, 2016 2:37 PM, "Michael Marineau" <notifications@github.com>
wrote:

> In locksmithctl/daemon.go
> <https://github.com/coreos/locksmith/pull/101#discussion_r52537948>:
>
> >  	if strategy == StrategyOff {
> > -		fmt.Fprintf(os.Stderr, "Reboot strategy is %q - shutting down.\n", strategy)
> > +		dlog.Warningf("Reboot strategy is %q - shutting down.", strategy)
>
> Shouldn't be a warning, iirc only cloudinit tries to interpret strategy
> itself and mask the unit if it is off. We shouldn't warn the user because
> they aren't using our old cloudinit magic. notice should do fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/locksmith/pull/101/files#r52537948>.
>